android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt

```diff
@@ -200,10 +201,25 @@ class ProfileViewModelTest : RobolectricTest() {
     every { listActionConfig.handleClickEvent(navController, resourceData) } just runs
     profileViewModel.onEvent(event)
     every { mockProfileViewModel.onEvent(any()) } just runs
-
     verifyAll {
       mockActionConfig.interpolate(any())
       profileViewModel.onEvent(event)
     }
   }
+
+  @Test
+  fun testThatChangeManagingEntityDisplaysBottomSheetWhenEligibleEntitiesExist() {
+    val event = mockk<ProfileEvent.OverflowMenuClick>(relaxed = true)
+    val managingEntity: ManagingEntityConfig = mockk(relaxed = true)
+
+    every { event.resourceData?.baseResourceType } returns ResourceType.Group
+    coEvery { registerRepository.loadResource<Group>(any()) } returns mockk(relaxed = true)
+    every { fhirPathDataExtractor.extractValue(any(), any()) } returns "true"
+
+    profileViewModel.changeManagingEntity(event, managingEntity)
+
+    coVerify { registerRepository.loadResource<Group>(any()) }
+    coVerify { fhirPathDataExtractor.extractValue(any(), any()) }
+    coVerify(exactly = 0) { profileViewModel.emitSnackBarState(any()) }
```

Comment:
Might be difficult, but is it possible to also check the bottom sheet? to ensure its shown.

If it is, we can skip that.

Reply:
Let me give it a shot